### ISSUE TITLE
Kubefile-linter or Clusterfile-linter for the community

### ISSUE THREAD

**Issue by allencloud** [sealerio/sealer#1353]
## Issue Description

Type: *feature request*

### Describe what feature you want

sealer provides a rule for users to edit a yaml file for building a cluster image. we named it as **Kubefile**, then Kubefile is a kind of protocol which user must obey. While Kubefile is a yaml file after all, it could happen to anybody that he typed in incorrect details in Kubefile. We wish to fail fast by some linter tools, rather than delivery incorrect content to the engine to process.

** I wish we could start up a brand new project like Kubefile-linter or something similar in https://github.com/sealerio helping users to improve effieciency.

Clusterfile has the same issue.

**Comment by kakaZhou719** [comment]
@allencloud , could we add a parameter to improve efficiency ,and it works like the Kubefile-linter.  like `sealer build --dryrun -f Kubefile -t myimages:v1`

**Comment by allencloud** [comment]
> @allencloud , could we add a parameter to improve efficiency ,and it works like the Kubefile-linter. like `sealer build --dryrun -f Kubefile -t myimages:v1`

@kakaZhou719 Of course we could do that just like what you suggested. While, I basically stand on the point that we should evolve the sealer project with the decoupled architecture. We designed the decoupled architecture and adopters could take advantages of each single part in different scenarios. After that, we could see that sealer is a group, not just a tool. Group has ecosystem, but tool may have limited scenario. We could grow sealer's community and ecosystem much better if we try the decoupled way.

In addition, we could open up several other repos, like preflight, status-checker, and some others.

**Comment by kakaZhou719** [comment]
yes, i agree, decoupled architecture will enrich our sealerio community. we could open up a new project to do the lint work, at the same time, sealer also can use it to do the pre-check.

**Comment by kakaZhou719** [comment]
meanwhile ,we could move application out from sealer project.

**Comment by allencloud** [comment]
> meanwhile ,we could move application and site doc out from the sealer project.

application and website should be first kinds to be separated.
